### PR TITLE
feat: most popular section added on create new ds page

### LINKED ISSUE #28550: [Feature]: Add new section on create datasource → Most Popular / Regroup and sort datasources
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

https://www.notion.so/appsmith/Task-breakdown-Start-with-data-787dd729e71e452abe14b46acae6e931?p=f4a4ee6631cf4d07bbd49e52c234d9a6&pm=s

### Why should this be worked on?

activation - onboarding revamp

### PR DESCRIPTION
**Pull request by sneha122**
## Description

This PR adds most popular datasources section on create new datasource page, This is added so that it will be easier for the user to check out most popular plugins at first. Currently list of most popular datasources has been hard coded to include Postgres, MySQL, MongoDB, Gsheets and REST API. We have gotten this data from mixpanel, on which plugins are used the most.

Screenshot 2023-11-14 at 6 06 15 PM

In future we may want to have this list vary based on each instance. 

#### PR fixes following issue(s)
Fixes #28550 
#### Type of change
- New feature (non-breaking change which adds functionality)
>
>
>
- [x] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by rohan-arthur** [comment]
client build is failing both for CI tests and DP generation. Moving this back to `in progress`

File: app/client/src/pages/Editor/IntegrationEditor/DatasourceHome.tsx [hunk 1]
```diff
@@ -215,17 +248,35 @@ class DatasourceHomeScreen extends React.Component<Props> {
...
+            </ApiCard>
+          ) : null}
         </DatasourceCardsContainer>
       </DatasourceHomePage>
     );
   }
 }
 
-const mapStateToProps = (state: AppState): ReduxStateProps => {
+const mapStateToProps = (state: AppState, props: any) => {
```
Comment (AmanAgarwal041) [inline app/client/src/pages/Editor/IntegrationEditor/DatasourceHome.tsx:273]: Can we change `any` to `ReduxStateProps & { createNewApiAction: ( pageId: string, from: EventLocation, apiType?: string ) => void; }` ?

File: app/client/src/pages/Editor/IntegrationEditor/DatasourceHome.tsx [hunk 2]
```diff
@@ -215,17 +248,35 @@ class DatasourceHomeScreen extends React.Component<Props> {
...
+              className="t--createBlankApiCard create-new-api"
+              onClick={() => this.handleOnClick()}
+            >
+              <CardContentWrapper data-testid="newapi-datasource-content-wrapper">
+                <img
+                  alt="New"
+                  className="curlImage t--plusImage content-icon"
+                  src={PlusLogo}
+                />
+                <p className="textBtn">REST API</p>
```
Comment (AmanAgarwal041) [inline app/client/src/pages/Editor/IntegrationEditor/DatasourceHome.tsx:262]: Can we use constants variable over here ?

**Review by AmanAgarwal041 — APPROVED** [review]

**Pull request merged**